QuantLib/ql/models/calibrationhelper.hpp

```diff
@@ -64,7 +71,7 @@ namespace QuantLib {
         virtual Real modelValue() const = 0;
 
         //! returns the error resulting from the model valuation
-        virtual Real calibrationError();
+        Real calibrationError() const;
```

Comment:
imo this is more a bug fix than a backward compatibility change (along with making CalibratedModel constructor private). We should just document in the new release the breaking fixes, as even boost does...